src/main/java/com/sphereon/factom/identity/did/json/CustomServiceTypeAdapter.java

```diff
@@ -24,7 +24,7 @@ public Service deserialize(JsonElement jsonElement, Type type, JsonDeserializati
                 .type(serviceMap.remove("type"))
                 .serviceEndpoint(serviceMap.remove("serviceEndpoint"));
         if (serviceMap.get("priorityRequirement") != null) {
-            service.priorityRequirement(Integer.valueOf(serviceMap.remove("priorityRequirement")));
+            service.priorityRequirement(Double.valueOf(serviceMap.remove("priorityRequirement")).intValue());
```

Comment:
Any reason for the Double -> .intValue?

Reply:
Yes, I was a bit stumped by this one. Aparently GSON deserializes all integers to `Number` which leaves the string value here "1.0" which will throw an exception if we use the `Integer.valueOf`.

Reply:
Yupz, makes sense. Please make a note for the why though, to ensure the next dev doesn't change it to something else, because the Double .intValue looks off.

Reply:
updated.